src/data-structures/hash-table.js

```diff
@@ -41,25 +48,71 @@
 
     for (i = 0; i < str.length; i += 1) {
       character = str.charCodeAt(i);
+      /*jshint -W016 */
       hashCode = ((hashCode << 5) - hashCode) + character;
       hashCode = hashCode & hashCode;
+      /*jshint -W016 */
     }
 
     return hashCode;
   };
 
-  exports.HashTable.prototype.put = function (key, value) {
+  exports.HashTable.prototype.put = function (key, data) {
     var hashCode = this.hashCode(key);
-    this.elements[hashCode] = value;
+    var newNode = new Node(key, data);
+
+    if (this.elements[hashCode] === null) {
+      this.elements[hashCode] = newNode;
+    } else {
+      var first = this.elements[hashCode];
+
+      while (first.next !== null) {
+        first = first.next;
+      }
+
+      first.next = newNode;
+      newNode.prev = first;
+    }
   };
 
   exports.HashTable.prototype.get = function (key) {
     var hashCode = this.hashCode(key);
-    return this.elements[hashCode];
+
+    if (this.elements[hashCode] === null) {
+      return null;
+    } else if (this.elements[hashCode].next === null) {
+      return this.elements[hashCode];
+    } else {
+      var first = this.elements[hashCode];
+
+      while (first.key !== key) {
```

Comment:
If the code enters the body of this if statement and the key is not found (i.e. `hasCode` has returned the same value for `key` like the one returned for different key during insertion) and you loop over the linked list, you may have condition where first equals `null`. In this case the code will throw an error. I believe you should also include the check:

```javascript
while (first && first.key !== key) {
  first = first.next;
}
```